modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -374,7 +374,13 @@ public class IgniteImpl implements Ignite {
         ReplicaService replicaSvc = new ReplicaService(clusterSvc.messagingService(), clock);
 
         // TODO: IGNITE-19344 - use nodeId that is validated on join (and probably generated differently).
-        txManager = new TxManagerImpl(replicaSvc, lockMgr, clock, new TransactionIdGenerator(() -> clusterSvc.nodeName().hashCode()));
+        txManager = new TxManagerImpl(
+                replicaSvc,
+                lockMgr,
+                clock,
+                new TransactionIdGenerator(() -> clusterSvc.nodeName().hashCode()),
+                () -> clusterSvc.topologyService().localMember().id()
```

Comment:
I think it's better to initialize the manager by a local member on start than pass the two closures through a constructor.

Reply:
Actually it is a closure to initialize lazy value. Passing the closure makes mocks more simple.
If you are speaking also about this `new TransactionIdGenerator(() -> clusterSvc.nodeName().hashCode()),` - here the closure is passed to `TransactionIdGenerator`, not to `TxManagerImpl` itself.